src/Agent/NewRelic/Agent/Core/Segments/Segment.cs

```diff
@@ -135,7 +135,7 @@ public void End()
                 var endTime = _transactionSegmentState.GetRelativeTime();
                 RelativeEndTime = endTime;
 
-                if (Agent.Instance.StackExchangeRedisCache != null && Agent.Instance.StackExchangeRedisCache.Count > 0)
+                if (Agent.Instance?.StackExchangeRedisCache != null && Agent.Instance.StackExchangeRedisCache.Count > 0)
```

Comment:
This seems unrelated to the rest of this PR. Does it belong here or in a separate PR?

Reply:
I might not be required at all.  My unit tests were passing and then out of the blue started to fail at this point with an object null error.  I made the change, restarted VS, and then it was all fine.  I kept it in since it made sense to check both for null, but it is probably safe to undo the change.

Reply:
Removed.